src/replicache.ts

```diff
@@ -572,23 +591,51 @@ export default class Replicache implements ReadTransaction {
    * mutations.
    */
   async push(): Promise<void> {
-    if (this._pushTimerId) {
-      clearTimeout(this._pushTimerId);
-      this._pushTimerId = 0;
-    }
+    this._clearPushTimer();
     await this._wrapInOnlineCheck(() => this._push(MAX_REAUTH_TRIES), 'Push');
   }
 
   private async _push(maxAuthTries: number): Promise<void> {
+    // Is there already a push in flight?
+    if (this._pushCounter > 0) {
+      // Here we create a promise/resolve pair that will resolve when the
+      // current push is done.
+      //
+      // This whole dance is to collapse incoming calls to push during an in
```

Comment:
Does any code anywhere rely on the promise returned from _push? Would it make it easier to have _push return void, not Promise<void>?

Reply:
`push` depends on it. We could have `push` return `void`. It would make tests harder to write but tests are tests and should not drive the API. 

Personally I think it is the right thing to have `push` return a Promise that resolves when the push is done.

Reply:
OK, not a huge thing.